### ISSUE TITLE
Feedback on the Waffle board

### ISSUE THREAD

**Issue by haadcode** [ipfs/blog#61]
If you allow me, I have some ideas how to improve the blog pipeline structuring on ipfs/blog Waffle. I'm writing this on the move and hope that tone is not too blunt or harsh. I'm writing this because I <3 you.

The current pipeline has a lot of steps atm that are not necessarily descriptive and purposeful. Think of a pipeline as something where items flow from step to step. Instead of "pushing" items to the next step, as to indicate "you need to work on these", the next step "pulls" the item from it's previous step to indicate "I'm working on this". And instead of indicating the status of each item with the steps, use labels to say "this item is ready to be pulled to the next step". I know this is hard to grasp initially, so allow me to give an example:

First, each blog post should be its own issue including ideas for new posts (eg. "Introducing Libp2p" should be one issue). All new items will go to the first step in the pipeline and thus (I think) an appropriate label gets assigned (eg. adding a new issues gets a label "Ideas" in the current setup) and if not, you can mark each new item with an appropriate label. This helps to keep all "need to check this" in one step and you don't need to separate "Ideas" from "meta" and also gives the needed information swiftly: "I see, this is an idea for a blog post whereas that one is bug" or "Ok, I need to pull this as it says it's ready to be reviewed".

Second, I would encourage to re-think the pipeline steps to be more generic, so that you can handle all issues in the repo as part of the pipeline, not just the writing tasks. Using common terminology from Agile process and Kanban boards, it could be something like this:

Backlog - All items that haven't been worked on yet and are not planned to be worked on yet. This is the default and entry step for every item/issue in the repo. We can label the items approriately to describe what they are: "bug", "idea", "meta", etc. Once items get approved, or to indicate that we should work on them, they should be bumped to the top of the column. The top items are the ones we should work on next, the ones at the bottom are ones that either haven't been reviewed yet or are low on priority. The prioritization can be done by whoever is the "owner" of the blog.

Current/TODO - Items here are the ones that will be worked on in the current sprint. The pull to this step can happen at the beginning of each sprint by the person who intends to work on it. By pulling an item to the next step, you commit to it: "I will work on this in the next n weeks". Ideally, we pull items that are on top of the list in the Backlog.

In Progress/WIP - Items that are currently and actively being worked on. When you start working on an item, you pull it from the "TODO" step to this step and by doing that, you say "I'm working on this _now_". Once an item is finished (or "done" according to the person who worked on it), the assignee can label it as "needs review" to indicate that the item can be pulled to the next column (and perhaps place it at the top of the column to indicate the urgency to pull it).

Review - Items that are currently in review. As soon as an item is marked as "need review" in the WIP step, the reviewer can pull it to this step to indicate that they are reviewing it _now_ (don't pull it to indicate "I will review this when I have time"). The outcome from this step can be either of two: the item needs more work or it's ready to be published. If an item needs re-work, the reviewer can pull it to the "Blocked" step and label it as "need rewrite" to indicate that this item can't move further down the pipeline until review feedback has been addressed. If the item is ready to be published, the reviewer can label it as "ready for publishing" and whoever is responsible for the publishing backlog/pressing "publish" button, can pull it to the approriate step.

Blocked - Items that are blocked to move down the pipeline in any way. This is to indicate everyone that there's a problem with this item and it needs urgent attention. Eg. if an item is blocked and has a label "need rewrite", one can tell that it has been reviewed but changes are needed before it can be published. Whoever was the assignee for the item, can then pull it back to "In Progress/WIP" as soon as they start working on the re-write.

To Be Published - Backlog of blog posts to be published. Items here are waiting to get pulled to the very last step in the pipeline. I reckon this could be optional and we can keep the to-be-published items in the "Review" step with a label "to be published" and bump them on top of the list to indicate the order of publishing.

Done - Items that have been fully completed: published blog posts, fixed bugs, etc. Items that end up here can never go backwards in the pipeline (ideally, exceptions might happen). End of the line.

To recap: simplify and generalize the pipeline steps to indicate active work vs. to-be-worked-on vs. blocked, use label to describe the item "types" and status.

Let me know if you have any questions or need clarification on any of this.

**Comment by jbenet** [comment]
I'll respond more fully later, but the point of the blog pipeline is to be very explicit about the steps needed to produce a blog post, to see where they get stuck and to get much better at estimating the time it takes to move them through the stages. it's about accurately and precisely capturing the blog writing process and the stages through it.

That's for the feedback, but there is missing context here, and we're talking about different things. this waffle board is an experiment to capture "the blog pipeline" and how to work with items moving through it. The goal is to be explicit about where a post is in its development lifetime, how it can move forward, and to give a very clear view on what's getting stuck towards publishing (explicitly what stage). The board's goal is not for dealing with arbitrary things on this repo, nor for capturing a sprint oriented process (working on this this week / blocked / etc) . It's a very different use case.

**Comment by jbenet** [comment]
It's capturing this: https://github.com/ipfs/blog/blob/master/PIPELINE.md

And to be clear: I agree with the "pull to be explicit that I'm working on this this week" but that's a _different_ process, and it's meta to this pipeline. This is not aiming to capture agile/kanban workflow, it's aiming to capture work product information.

**Comment by RichardLitt** [comment]
> We can have another board linked to the former. This is so that we can have two different label sets per pipeline, so we can do BOTH what @haad and @jbenet had in mind, 

> - one board for the STATE of the blog post
> - one board for who's working on what when

@haadcode What do you think about this? You're the expert on this process; would you like to set it up?

**Comment by mishmosh** [comment]
[Pipeline.md](https://github.com/ipfs/blog/blob/master/PIPELINE.md) is the current process. It's not perfect, but works reasonably well.

Rather than redesign process right now, let's start by shipping blog posts every month as discussed in #101. We can revisit process if needed.
